assets/js/search.js

```diff
@@ -0,0 +1,211 @@
+
+    var searchResults = document.getElementById('search-results');
+    var searchResultsWeb = document.getElementById('search-results-web');
+    var searchResultsDiv = document.getElementById('searchResultsDiv');
+
+    if (results.length) { // Are there any results?
+
+        searchResultsDiv.classList.add("active");
+        var appendString = '';
+
+        for (var i = 0; i < results.length; i++) {  // Iterate over the results
+            var item = links[results[i].ref];
+
+            let title = item.displayName || item.shortDescription
+
+            appendString += '<li><div class="searchResultItem"><span class="searchResultTitle">' + title + '</span>';
+
+            const siteUrl = new URL(window.location.href);
+            const permalink = `${siteUrl.origin}#${item.name}`
+
+            appendString += `<div class="searchResultLinks"><a href="${item.url}" onclick="return clearInput();">קח אותי ליוזמה</a> / <a onclick="searchResultClicked(${item.name})">עוד מידע</a></div></div></li>`;
+            
+        }
+
+        searchResults.innerHTML = appendString;
+        searchResultsWeb.innerHTML = appendString;
+    } else {
+        searchResults.innerHTML = '<li>No results found</li>';
+        searchResultsWeb.innerHTML = '<li>No results found</li>';
+        searchResultsDiv.classList.remove("active");
+    }
+}
+
+function prepareIndex() {
+
+    let linksOnly = {}
+    var idx = lunr(function () {
+
+        this.use(lunr.multiLanguage('en', 'he'));
+
+        this.field('displayName');
+        this.field('description');
+        this.field('shortDescription');
+
+        window.israelLinks.forEach((category)=>{ // Add the data to lunr
+
+            const CategoryEntry = {
+                id: category.name,
+                name: category.name,
+                displayName: category.displayName,
+                description: category.description,
+                subCategories: category.subCategories,
+            }
+
+            category.subCategories.forEach((subCategory)=>{
+                const subCategoryEntry =  {
+                    id: subCategory.name,
+                    displayName: subCategory.displayName,
+                }
+
+                subCategoryEntry.links = subCategory.links.forEach((link)=> {
+                    const linkEntry = {
+                        id: link.name,
+                        displayName: link.displayName,
+                        shortDescription: link.shortDescription,
+                        description: link.description
+                    }
+
+                    linksOnly[link.name] = link
+                })
+            })
+        })
+
+        Object.values(linksOnly).forEach((link)=> {
+            const linkEntry = {
+                id: link.name,
+                displayName: link.displayName,
+                shortDescription: link.shortDescription,
+                description: link.description
+            }
+    
+            this.add(linkEntry);
```

Comment:
Do we want to add the `displayName` and `description` of each link's category and subCategory to `linkEntry`? This way, when looking for `אוכל`, the results list won't be empty.